src/Adapter/Product/CommandHandler/UpdateProductPricesHandler.php

```diff
@@ -0,0 +1,279 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\CommandHandler;
+
+use PrestaShop\Decimal\Number;
+use PrestaShop\PrestaShop\Adapter\Entity\TaxRulesGroup;
+use PrestaShop\PrestaShop\Adapter\Product\AbstractProductHandler;
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\UpdateProductPricesCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\CommandHandler\UpdateProductPricesHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\CannotUpdateProductException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductException;
+use PrestaShop\PrestaShop\Core\Domain\Product\ProductTaxRulesGroupSettings;
+use PrestaShop\PrestaShop\Core\Util\Number\NumberExtractor;
+use PrestaShopException;
+use Product;
+
+/**
+ * Updates product price information using legacy object models
+ */
+final class UpdateProductPricesHandler extends AbstractProductHandler implements UpdateProductPricesHandlerInterface
+{
+    /**
+     * @var NumberExtractor
+     */
+    private $numberExtractor;
+
+    /**
+     * @var Product the product being updated
+     */
+    private $product;
+
+    /**
+     * @param NumberExtractor $numberExtractor
+     */
+    public function __construct(
+        NumberExtractor $numberExtractor
+    ) {
+        $this->numberExtractor = $numberExtractor;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(UpdateProductPricesCommand $command): void
+    {
+        $product = $this->getProduct($command->getProductId());
+        $this->product = $product;
+        $this->fillUpdatableFieldsWithCommandData($product, $command);
+        $product->setFieldsToUpdate($this->fieldsToUpdate);
+
+        if (empty($this->fieldsToUpdate)) {
+            return;
+        }
+
+        $this->performUpdate($product);
```

Comment:
Since `$this->fieldsToUpdate` is now in the abstract handler, and we will ned the same principle in other handlers, maybe all this could be moved into `performUpdate` and the method can be put in the abstract handler as well

And to finalize the feature add a `markUpdatedField` to avoid modifying the array directly and have a more semantic function

Of course it's just a suggestion, not blocking for this PR and can be done in another PR, along with the harmonization on other handlers